tests/devapps/UWP/MainPage.xaml.cs

```diff
@@ -31,12 +33,46 @@ public sealed partial class MainPage : Page
         private readonly static string ClientID = "9058d700-ccd7-4dd4-a029-aec31995add0";
         private readonly static string Authority = "https://login.microsoftonline.com/common/";
         private readonly static IEnumerable<string> Scopes = new[] { "https://graph.windows.net/.default" };
+        private const string Resource = "https://graph.windows.net";
+
 
         public MainPage()
         {
             this.InitializeComponent();
 
             _pca = new PublicClientApplication(ClientID, Authority);
+
+            // custom serialization
+            _pca.UserTokenCache.SetAfterAccess((tokenCacheNotifcation) =>
+            {
+                if (tokenCacheNotifcation.HasStateChanged)
+                {
+                    var cacheFile = ApplicationData.Current.LocalFolder.CreateFileAsync("msal_cache.txt", CreationCollisionOption.ReplaceExisting)
+                        .AsTask()
+                        .GetAwaiter()
+                        .GetResult();
+
+                    File.WriteAllBytes(cacheFile.Path, tokenCacheNotifcation.TokenCache.Serialize());
+
+                }
+
+
+            });
+
+            _pca.UserTokenCache.SetBeforeAccess((tokenCacheNotifcation) =>
+            {
+                var cacheFile = (ApplicationData.Current.LocalFolder.TryGetItemAsync("msal_cache.txt"))
```

Comment:
Does this replace the locks?

Reply:
What locks are you referring to ? All token cache operations are thread safe because they are already inside locks. Would you like to protect against multiple apps writing to the same file?

Reply:
the ADAL samples had a lock. I understand it's no longer needed with MSAL.
Actually you also make a good point @bgavrilMS : how can we protect against multiple apps writing to the same file?

Reply:
We have the same challenge for other places than UWP. Something to solve for the dev tools and customers in general. A common solution can possibly be found to that challenge (which may have to vary based on OS).

Reply:
No locks needed, this code is called from a single thread context